src/wamr/wamr.cc

```diff
@@ -0,0 +1,654 @@
+  };
+  FOR_ALL_WASM_VM_IMPORTS(_REGISTER_HOST_FUNCTION)
+#undef _REGISTER_HOST_FUNCTION
+
+#define _GET_MODULE_FUNCTION(T)                                                                    \
+  void getFunction(std::string_view function_name, T *f) override {                                \
+    getModuleFunctionImpl(function_name, f);                                                       \
+  };
+  FOR_ALL_WASM_VM_EXPORTS(_GET_MODULE_FUNCTION)
+#undef _GET_MODULE_FUNCTION
+private:
+  template <typename... Args>
+  void registerHostFunctionImpl(std::string_view module_name, std::string_view function_name,
+                                void (*function)(void *, Args...));
+
+  template <typename R, typename... Args>
+  void registerHostFunctionImpl(std::string_view module_name, std::string_view function_name,
+                                R (*function)(void *, Args...));
+
+  template <typename... Args>
+  void getModuleFunctionImpl(std::string_view function_name,
+                             std::function<void(ContextBase *, Args...)> *function);
+
+  template <typename R, typename... Args>
+  void getModuleFunctionImpl(std::string_view function_name,
+                             std::function<R(ContextBase *, Args...)> *function);
+
+  WasmStorePtr store_;
+  WasmModulePtr module_;
+  WasmInstancePtr instance_;
+
+  WasmMemoryPtr memory_;
+  WasmTablePtr table_;
+
+  std::unordered_map<std::string, HostFuncDataPtr> host_functions_;
+  std::unordered_map<std::string, WasmFuncPtr> module_functions_;
+  AbiVersion abi_version_;
+};
+
+bool Wamr::load(const std::string &code, bool allow_precompiled) {
+  store_ = wasm_store_new(engine());
+
+  // Get ABI version from bytecode.
+  if (!common::BytecodeUtil::getAbiVersion(code, abi_version_)) {
+    fail(FailState::UnableToInitializeCode, "Failed to parse corrupted Wasm module");
+    return false;
+  }
+
+  std::string stripped;
+  if (!common::BytecodeUtil::getStrippedSource(code, stripped)) {
+    fail(FailState::UnableToInitializeCode, "Failed to parse corrupted Wasm module");
+    return false;
+  };
+
+  WasmByteVec stripped_vec;
+  wasm_byte_vec_new(stripped_vec.get(), stripped.size(), stripped.data());
+  module_ = wasm_module_new(store_.get(), stripped_vec.get());
+
+  return module_ != nullptr;
+}
+
+std::unique_ptr<WasmVm> Wamr::clone() {
+  assert(module_ != nullptr);
+
+  auto clone = std::make_unique<Wamr>();
+
+  clone->integration().reset(integration()->clone());
+
+  clone->store_ = wasm_store_new(engine());
+
+  std::string stripped;
+  if (!common::BytecodeUtil::getStrippedSource("", stripped)) {
+    fail(FailState::UnableToInitializeCode, "Failed to parse corrupted Wasm module");
+    return nullptr;
+  };
+
+  WasmByteVec stripped_vec;
+  wasm_byte_vec_new(stripped_vec.get(), stripped.size(), stripped.data());
+  clone->module_ = wasm_module_new(store_.get(), stripped_vec.get());
+
+  clone->abi_version_ = abi_version_;
+
+  return clone;
```

Comment:
This is not how runtime with `Cloneable::CompiledBytecode` should behave. If you're creating a new instance from scratch, then you should return `Cloneable::NotCloneable` and not implement the `clone()` function.

Reply:
Fixed